auth/client/src/main/java/org/wildfly/security/credential/source/OAuth2CredentialSource.java

```diff
@@ -292,6 +300,105 @@ public Builder clientCredentials(String id, String secret) {
             return this;
         }
 
+        /**
+         * <p>Configure OAuth2 Resource Owner Masked Password Grant Type as defined by the OAuth2 specification.
+         *
+         * <p>When using this grant type, make sure to also configure one of the supported client authentication methods. For instance,
+         * make sure to provide client credentials via {@link #clientCredentials(String, String)}.
+         *
+         * @param userName the resource owner's user name
+         * @param maskedPassword the masked password, as a string (must not be {@code null})
+         * @param algorithm the algorithm (can be {@code null}, default:"masked-MD5-DES")
+         * @param initialKeyMaterial the initial key material, as a string(can be {@code null}, default:"somearbitrarycrazystringthatdoesnotmatter")
+         * @param iterationCount the iteration count, as an integer (must not be less than 1)
+         * @param salt the salt, as a string (must not be {@code null})
+         * @param initializationVector the initialization vector, as a string (can be {@code null})
+         * @return this instance.
+         */
+        public Builder useResourceOwnerMaskedPassword(String userName, String maskedPassword, String algorithm, String initialKeyMaterial, int iterationCount, String salt, String initializationVector) throws NoSuchAlgorithmException, InvalidKeySpecException {
+
+            Assert.assertNotNull(maskedPassword);
+            Assert.checkMinimumParameter("iterationCount", 1, iterationCount);
+            Assert.assertNotNull(salt);
+
+            byte[] finalMaskedSecretBytes = CodePointIterator.ofString(maskedPassword).base64Decode().drain();
+            if (algorithm == null) algorithm = MaskedPassword.ALGORITHM_MASKED_MD5_DES;
+            char[] finalInitialKeyMaterial = initialKeyMaterial == null ? "somearbitrarycrazystringthatdoesnotmatter".toCharArray() : initialKeyMaterial.toCharArray();
```

Comment:
It would be good to make ""somearbitrarycrazystringthatdoesnotmatter" a constant.

Reply:
Added